packages/docusaurus/src/server/getHostPort.ts

```diff
@@ -98,6 +98,9 @@ export async function getHostPort(options: HostPortOptions): Promise<{
 }> {
   const host = options.host ?? 'localhost';
   const basePort = options.port ? parseInt(options.port, 10) : DEFAULT_PORT;
-  const port = await choosePort(host, basePort);
+  const port = process.env.PORT
+    ? await choosePort(host, parseInt(process.env.PORT, 10))
```

Comment:
not a big deal but what about moving process.env.PORT to `constants.ts`, we already use a few env variables there to init default so it looks better for consistency